internal/utils/utils.go

```diff
@@ -0,0 +1,47 @@
+package utils
```

Comment:
There's a few common functions that are fairly important, so I'd like to isolate them into a `utils` package and add proper tests.